charts/redpanda/chart_test.go

```diff
@@ -95,11 +96,26 @@ func TestTemplate(t *testing.T) {
 	client, err := helm.New(helm.Options{ConfigHome: testutil.TempDir(t)})
 	require.NoError(t, err)
 
+	lock, err := helm.GetChartLock("Chart.lock")
+	require.NoError(t, err)
+
 	// Chart deps are kept within ./charts as a tgz archive. Helm dep build
 	// will ensure that ./charts is in sync with Chart.lock.
 	_, err = exec.CommandContext(ctx, "helm", "dep", "build").CombinedOutput()
 	require.NoError(t, err, "failed to refresh helm dependencies")
 
+	newLock, err := helm.GetChartLock("Chart.lock")
+	require.NoError(t, err)
+
+	// Comparison between Chart.lock before and after `helm dep build` execution is required to prevent
+	// circular dependency between Chart.lock update, which is included in release [Auto commit], and
+	// next unnecessary release pipeline execution. Allowed change in Chart.lock is only when dependencies
+	// are different. Dependencies might change only when Redpanda chart dependencies got new release.
+	if reflect.DeepEqual(lock.Dependencies, newLock.Dependencies) {
```

Comment:
nit: I think we could comparing .Digest here would be equivalent and slightly easier to reason about given it's just a string.

For future reference, it might be good to have this comment explicitly call out that `Generated` is the problem.

Reply:
Thanks, I changed to `Digest`.

I added information about `generated` filed being to culprit in comment.